src/main/java/org/humancellatlas/ingest/project/ProjectNotifications.java

```diff
@@ -0,0 +1,119 @@
+package org.humancellatlas.ingest.project;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.humancellatlas.ingest.notifications.NotificationService;
+import org.humancellatlas.ingest.notifications.exception.DuplicateNotification;
+import org.humancellatlas.ingest.notifications.model.Checksum;
+import org.humancellatlas.ingest.notifications.model.Notification;
+import org.humancellatlas.ingest.notifications.model.NotificationRequest;
+import org.humancellatlas.ingest.state.ValidationState;
+import org.humancellatlas.ingest.user.IdentityService;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Component;
+import org.springframework.util.DigestUtils;
+
+@Component
+@RequiredArgsConstructor
+public class ProjectNotifications {
```

Comment:
It's a bit strange for me to see plural nouns as class names. This could be renamed as ProjectEventHandler, ProjectNotificationSender or something similar.

Reply:
Will rename this to ProjectEventHandlers